### ISSUE TITLE
Scratch buffer size on AMD's implementation

### ISSUE THREAD

**Issue by AaronHaganAMD** [NVIDIA/Q2RTX#99]
Would it be possible to up the scratch buffer size from:

`#define SIZE_SCRATCH_BUFFER (1 << 24)`

to 

`#define SIZE_SCRATCH_BUFFER (1 << 25)`

We have noticed we occasionally hit the following assert in a few scenes / scenarios.
`assert(scratch_buf_ptr < SIZE_SCRATCH_BUFFER);`

**Comment by apanteleev** [comment]
No problem, done.

Please also take a look at the crashes in your driver that happen in `vkCreateRayTracingPipelinesKHR` when some seemingly random changes are introduced to the RT shaders, and that cause e.g. these reverts: https://github.com/NVIDIA/Q2RTX/commit/ae06b4dba2e37e20a7599c84e685c014469f3dae https://github.com/NVIDIA/Q2RTX/commit/12e7878c0caa4efce35f32bd487d08868446a4b5

**Comment by AaronHaganAMD** [comment]
Thank you for updating the scratch buffer size. 
I reverted those reverts you mentioned, rebuilt and the pipelines seem to compile ok for me... 
If you are still seeing it crash can you create a branch with the failing pipeline code and I would be happy to take a look at it.

**Comment by apanteleev** [comment]
Hi Aaron, I've updated the AMD GPU driver to the latest available version (20.12.2) and still see those crashes. It actually became worse. Right after updating, it was crashing on both the released v1.4.0 (aed6d4f71479af4f0073f558069f9773c7642c1e) and on the ToT master (0a16c7302d19e884a9fa6e46044cd69f74b60469). Cleaning the driver with DDU and installing it again helped with 1.4.0 but not master. Below is the relevant part of the crash log.

Let me know if I can be more helpful in debugging this. We can also switch to email - please ping me at alpanteleev @ nvidia . com

```
0x00007ffee0320000 0x00007ffee29c9000 amdvlk64 (version 27.20.14501.33001, symbols export) *

Exception information:
Code: 0xc0000005
Address: 0x00007ffee09ca69e (amdvlk64)

Thread context:
RIP: 0x00007ffee09ca69e RBP: 0x0000000000000000 RSP: 0x00000037c49d9500
RAX: 0x0000000000000000 RBX: 0x0000018e39c30f98 RCX: 0x0000018e39c31208
RDX: 0x0000018e3b0a77d8 RSI: 0x0000000000000001 RDI: 0x0000018e39c31208
R8 : 0x0000000000000000 R9 : 0x00007ffee0320000 R10: 0x0000018e3e84a680
R11: 0x0000018e600d3b28 R12: 0x0000000000000001 R13: 0x0000018e3bbb9258
R14: 0x0000000000000000 R15: 0x0000000000000000

Stack trace:
0: 0x0000000000000000 0x0000018e5fe964b8 0x0000000000000000 0x0000000000000000 amdvlk64!0x00007ffee09ca69e
1: 0x0000018e55aa8538 0x0000018e55aa8538 0x0000018e5fe964b8 0x0000018e60ad7330 amdvlk64!0x00007ffee0a7050b
2: 0x0000018e600d3b40 0x0000018e60af97a0 0x0000018e5fe964b8 0x0000018e600d3b50 amdvlk64!0x00007ffee07fbe77
3: 0x0000018e5fe964b8 0x0000018e60b5e660 0x0000018e60af97a0 0x0000018e5fe964b8 amdvlk64!0x00007ffee081b02c
4: 0x0000018e5fe964b8 0x0000018e60af97a0 0x0000018e60af1b20 0x0000018e5fe964b8 amdvlk64!0x00007ffee080026b
5: 0x0000000000000004 0x00000037c49da581 0x0000018e60af1878 0x0000018e60af1b20 amdvlk64!0x00007ffee05a5717
6: 0x00000037c49da8a0 0x00000037c49da640 0x00000037c49da8a0 0x0000018e5a62f519 amdvlk64!0x00007ffee0753120
7: 0x0000000000000000 0x00000037c49da870 0x00000037c49da870 0x00000037c49da900 amdvlk64!0x00007ffee061ac06
8: 0x0000000000000000 0x0000018e5a621590 0x00000037c49da860 0x0000000000000000 amdvlk64!0x00007ffee0619fdd
9: 0x00000037c49dc560 0x0000018e00000004 0x0000000000000000 0x00000037c49dc2b0 amdvlk64!0x00007ffee05a326b
10: 0x0000018e55b8dc78 0x0000018e5a6fba98 0x0000000000000000 0x00000037c49dbef0 amdvlk64!0x00007ffee03ae91f
11: 0x0000000000000001 0x0000018e5a6fba98 0x0000000000000000 0x0000000000000000 amdvlk64!0x00007ffee035d082
12: 0x0000018e5fec8540 0x0000000000000000 0x00000037c49df210 0x0000018e55b0e3d0 amdvlk64!0x00007ffee05672e7
13: 0x0000000000000000 0x0000000000000000 0x0000000000000000 0x00007ffe00000001 amdvlk64!0x00007ffee0365a72
14: 0x0000000000000000 0x0000000000000000 0x0000000000000000 0xcccccccccccccccc q2rtx!vkpt_pt_create_pipelines+0x00000a77
15: 0x00007ff700000000 0x00007ff72e734df4 0xcccccccc00000000 0xcccccccccccccccc q2rtx!vkpt_initialize_all+0x0000012f
16: 0x0000000000000001 0x00007ff72e61f640 0xcccccccc00000001 0xcccccccccccccccc q2rtx!R_Init_RTX+0x00000d36
```

**Comment by AaronHaganAMD** [comment]
I believe this issue can be closed now.
